Don't show `const` in docs when it's not available

quodlibetor opened the pull request:
Fixes #31098 

AFAICT this is the only place where rustdoc explicitly checks if we are on stable before emitting content, so I can't tell if this is the sane way to handle this, or if anything else should be done to make sure that nobody forgets to remove this check when `const` is stabilized.

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @alexcrichton (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. Due to the way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [the contribution instructions](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

alexcrichton commented:
Does this break rendering other parts like `*const T`? I think the helper here may be used in more than one spot

quodlibetor commented:
It might, although I can't find it in `render.rs` and all the uses of `hir::Constness` that I could find were in building methods or functions, but I'll double check that some `*cont T` still shows up in the output.

Originally when I implemented this I added const filtering to the "build function or method" fns, but that required lying about the constness in the struct. If filtering in the render helper doesn't work out do you have an opinion on what I should do? It's not *really* lying about the constness of an object if it can't be made const in stable, is it?

alexcrichton commented:
After checking it looks like this is indeed only used for `const fn`, but I'd personally feel a little more comfortable at least if this were a bit more of a targeted conditional. Could this conditional be placed directly where `const fn` is being handled because that's what this is actually trying to apply to?

Also can this use the `get_unstable_features_setting` function in the compiler instead of reading the environment variable manually?

quodlibetor commented:
Sort of like so?

alexcrichton commented:
@bors: r+ d59372b7aad364a287f21cfbcf12e85d8be97fc7

Indeed, thanks!

quodlibetor commented:
Awesome! First rust PR :)
